frontend/src/store/bucketStore.ts

```diff
@@ -1,199 +1,95 @@
-import { ref, isProxy, toRaw } from 'vue';
 import { defineStore, storeToRefs } from 'pinia';
-import { useToast } from 'primevue/usetoast';
-
-import { bucketService, permissionService, userService } from '@/services';
-import { useConfigStore, useUserStore } from '@/store';
-import { Permissions } from '@/utils/constants';
+import { computed, ref, unref } from 'vue';
+import { useToast } from '@/lib/primevue';
+import { bucketService } from '@/services';
+import { useAppStore, usePermissionStore, useUserStore } from '@/store';
+import { partition } from '@/utils/utils';
 
 import type { Ref } from 'vue';
-import type { Bucket, IdentityProvider, Permission, User, UserPermissions } from '@/interfaces';
+import type { Bucket, BucketPermissionsOptions } from '@/types';
 
 export const useBucketStore = defineStore('bucket', () => {
-  const { getConfig } = useConfigStore();
-  const { currentUser } = storeToRefs(useUserStore());
   const toast = useToast();
 
+  // Store
+  const appStore = useAppStore();
+  const permissionStore = usePermissionStore();
+  const { currentUser } = storeToRefs(useUserStore());
```

Comment:
We should make a pass on this and make sure to change this to computed equivalents so that we're not crossing the states of our bucket definitions.